tidalapi/session.py

```diff
@@ -431,6 +458,109 @@ def login_oauth_file(self, oauth_file: Path) -> bool:
             log.info("TIDAL Login KO")
             return False
 
+    def login_pkce(self, fn_print: Callable[[str], None] = print) -> None:
+        """Login handler for PKCE based authentication. This is the only way how to get
+        access to HiRes (Up to 24-bit, 192 kHz) FLAC files.
+
+        This handler will ask you to follow a URL, process with the login in the browser
+        and copy & paste the URL of the redirected browser page.
+
+        :param fn_print: A function which will be called to print the instructions,
+            defaults to `print()`.
+        :type fn_print: Callable, optional
+        :return:
+        """
+        # Get login url
+        url_login: str = self.pkce_login_url()
+
+        fn_print("READ CAREFULLY!")
+        fn_print("---------------")
+        fn_print(
+            "You need to open this link and login with your username and password. "
+            "Afterwards you will be redirected to an 'Oops' page. "
+            "To complete the login you must copy the URL from this 'Oops' page and paste it to the input field."
+        )
+        fn_print(url_login)
+
+        # Get redirect URL from user input.
+        url_redirect: str = input("Paste 'Ooops' page URL here and press <ENTER>:")
+        # Query for auth tokens
+        json: dict[str, Union[str, int]] = self.pkce_get_auth_token(url_redirect)
+
+        # Parse and set tokens.
+        self.process_auth_token(json)
+
+        # Swap the client_id and secret
+        #self.client_enable_hires()
```

Comment:
@exislow Any reason this was not enabled in this PR?

Reply:
It works with and without this. To avoid a possible break of existing functions I have decided to comment it out. But if you are sure, that all tests are running correct with an OAuth2 token you could also enable line 494. If you like I can also get you a pull request for this. Just let me know, how you decide.